src/model/model.ts

```diff
@@ -252,12 +277,19 @@ export class Model {
       if (!ast) {
         return false;
       }
-      // const index = ast.policy.indexOf(oldRule);
-      const index = ast.policy.findIndex((r) => util.arrayEquals(r, oldRule));
-      if (index !== -1) {
-        ast.policy[index] = newRule;
-        return true;
+
+      const priorityFlag = ast.tokens.indexOf('p_priority') !== -1;
+
+      if (priorityFlag) {
+        this.removePolicy(sec, ptype, oldRule);
+        this.addPolicy(sec, ptype, newRule);
+      } else {
+        const index = ast.policy.findIndex((r) => util.arrayEquals(r, oldRule));
+        if (index !== -1) {
+          ast.policy[index] = newRule;
```

Comment:
updatePolicy should only update rule with same priority.

Reply:
So it means if update got a policy with different priority, it should throw an error?

Reply:
this is my opinion. @hsluoyz what do you think that?

Reply:
Update priority currently didn't be supported in GoLang version. This will be the standard.

Reply:
@Zxilly why do this?

```js
this.removePolicy(sec, ptype, oldRule);
this.addPolicy(sec, ptype, newRule);
```

Why can't we treat `priority` policy element as a normal one like other elements? I think the order in the policy list doesn't matter, right? The `Enforce()` call will consider the `priority` sorting.

Reply:
explicit priority needs the policies sort by the number of priority. If updatepolicy() got a rule with different priority, it will break the order.
Should updatepolicy() only receive the newRule which have the same priority with oldRule?

Reply:
@Zxilly if the priority has changed, then insert it into the right place, or append to the list rear, and sort the list.

Reply:
@hsluoyz sort the whole list will lead to slow respond if we have a large number policy. I will find a better solution.

Reply:
@techoner @hsluoyz I add a commit refer to this discussion. plz review again.

Reply:
@Zxilly LGTM. 
@hsluoyz could you review this changes?